game/game.go

```diff
@@ -562,6 +562,62 @@ func (g *Game) performMove(m *move) (bool, string) {
 	return true, tgtPiece
 }
 
+// Some logic is duplicated with the function performMove. Look into reusing logic where possible.
+func forecastMove(currentBoard [][]string, pieceToInt map[string]int, canAttack [][]bool, srcFile, srcRank, tgtFile, tgtRank int) (bool, [][]string) {
```

Comment:
Could you change this to use [named return values](https://tour.golang.org/basics/7)?  Feel free to update the other function for consistency.